src/test/tests.ts

```diff
@@ -809,15 +805,16 @@ describe('Server', function () {
 
     setTimeout(() => {
       expect(connectionCallbackSpy.calledOnce).to.be.true;
-      expect(connectionCallbackSpy.getCall(0).args[0]).to.equal('Error');
+      expect(connectionCallbackSpy.getCall(0).args[0].message).to.equal('Error');
```

Comment:
nope :)
this will break it, that's a specific error you are handling there and that's the connection one.
so you need to do it on legacy mapping function.
then yeah, also add some flag to enforce legacy as an option.
then we will be able to add tests for mixing them 👍